pubtools/_pulp/tasks/publish.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import os
+import sys
+import re
+from datetime import datetime
+from functools import partial
+from more_executors.futures import f_map, f_sequence
+
+from pubtools.pulplib import Criteria, Matcher, PublishOptions
+
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.services import (
+    PulpClientService,
+    FastPurgeClientService,
+    UdCacheClientService,
+)
+
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+def publish_date(str_date):
+    # validates publish-before date
+    return datetime.strptime(str_date, "%Y-%m-%d")
+
+
+class Publish(
+    PulpClientService, FastPurgeClientService, UdCacheClientService, PulpTask
+):
+    """Publishes the pulp repositories to the endpoints defined by the distributors
+
+    This command will publish the pulp repositories provided in the request or
+    fetched using the filters(url-regex or published-before) or an intersection
+    of input repos and filters.
+    """
+
+    def add_args(self):
+        super(Publish, self).add_args()
+
+        self.parser.add_argument(
+            "--repos", help="list of repos to be published", nargs="+"
```

Comment:
maintenance-on/maintenance-off need the same arguments (select a set of existing repos), but they're named differently there:

```
                                    [--repo-url-regex REPO_URL_REGEX]
                                    [--repo-ids REPO_IDS [REPO_IDS ...]]
```

Could you please be consistent with it?  I don't think it makes sense to have one command taking ``--repo-ids`` and another taking ``--repos`` for the same purpose.

Reply:
yeah. updated this.